gqlgen.yml

```diff
@@ -3,11 +3,15 @@
 # Refer to https://gqlgen.com/config/
 # for detailed .gqlgen.yml documentation.
 
-schema: graphql/schema.graphql
+schema:
+  - graphql/schema.graphql
 exec:
   filename: graphql/generated.go
 model:
   filename: graphql/models_gen.go
 resolver:
   filename: graphql/resolvers.go
   type: Resolver
+models:
+  Patch:
+    model: github.com/evergreen-ci/evergreen/model/patch.Patch
```

Comment:
@john-m-liu might have an opinion here. 

I'm somewhat dubious about using this rather than the rest model. I don't think that we should be referncing the db models.

Reply:
i think it has to be the DB model. the resolver uses these types to understand how the DB model relates to resolver return signature. the resolvers make database calls and need to know how the raw data from the DB compares to the return type. For example, if description is a string in database model and a string in the schema then it knows the description field does not need a custom resolver. the API model, on the other hand, is not representative of the data that comes from the DB and would result is incorrect codegen.

Reply:
You can think of the gqlgen code gen as the API models. It takes a DB model, takes the final form it's supposed to be in (the schema) and resolves them.

Reply:
This should be the API model. The generator doesn't know the difference - this mapping is just for the generator to reuse resolvers for types that it knows about already

Reply:
I've become pretty convinced that we shouldn't be using the DB models here.

Reply:
We did some investigation - the generator gets confused by APIString and APITime and doesn't resolve the type alias to pointers to primitive types. Because of that and the fact that this will probably be changed in the near future, let's use the db model for now and open a ticket to change this and figure out a solution that will allow us to use the API models going forward

Reply:
here is ticket to change all API models to be compatible with gqlgen: https://jira.mongodb.org/secure/RapidBoard.jspa?rapidView=61&view=planning&selectedIssue=EVG-7161&epics=visible

Reply:
I'm super hesitant to acquire technical debt like this in greenfield projects like this one.